File: backend/Backend/Models/Controller.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    public class Controller
    {
        [Key]
        [Column(TypeName = "nvarchar(100)")]
        public string ControllerId { get; set; } = string.Empty;

        // Alias cho UI: trả về thêm field "id" = controllerId
        public string Id => ControllerId;

        public string Ip { get; set; } = string.Empty;

        public int Port { get; set; }

        [Column(TypeName = "nvarchar(100)")]
        public string SerialNumber { get; set; } = string.Empty;
    }
}

File: lib/models/controller.dart
class Controller {
  String controllerId;
  String ip;
  int port;
  String serialNumber;

  Controller({
    required this.controllerId,
    required this.ip,
    required this.port,
    required this.serialNumber,
  });

  factory Controller.fromJson(Map<String, dynamic> json) {
    return Controller(
      controllerId: json['controllerId'] as String,
      ip: json['ip'] as String,
      port: json['port'] as int,
      serialNumber: json['serialNumber'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'controllerId': controllerId,
      'ip': ip,
      'port': port,
      'serialNumber': serialNumber,
    };
  }
}
